### DISCUSSION TITLE
over, lens

### DISCUSSION THREAD

**Discussion by draconteus** [remeda/remeda#969] [Suggest a Function]
Hi,

Is there any Remeda implementation planned / alternative solution for `over` and `lens...` functions like you can find in Ramda?
We would like to move away from Ramda completely, but these are crucial functions for some transformation pipeline.

**Comment by cjquines** [comment]
i'm curious if this is still needed:

* for `lensProp` there's `prop` and `set`
* for `lensPath` there's `pathOr` and `setPath`
* for `lensIndex` , there's `a => a[i]` and `a => a.with(i, v)`

**Comment by agriffis** [comment]
I'm interested in this, too. I'm currently using Rambdax as a Ramda replacement, and it has lenses, but I'm intrigued by Remeda. Lenses are one of the things I would need to switch.

@cjquines Indeed lenses can be based on those functions, but lenses provide a composable abstraction of getter/setter pairs. There's some more in-depth explanation at https://www.schoolofhaskell.com/user/tel/lenses-from-scratch
